src/Routing/RouteSubscriber.php

```diff
@@ -19,20 +20,36 @@ final class RouteSubscriber extends RouteSubscriberBase {
    */
   private $entityTypeManager;
 
+  /**
+   * The module handler.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
   /**
    * RouteSubscriber constructor.
    *
    * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
-   *    The entity type manager.
+   *   The entity type manager.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, ModuleHandlerInterface $module_handler) {
     $this->entityTypeManager = $entity_type_manager;
+    $this->moduleHandler = $module_handler;
   }
 
   /**
    * {@inheritdoc}
    */
   protected function alterRoutes(RouteCollection $collection) {
+    // Only run if Field UI is installed.
+    if (!$this->moduleHandler->moduleExists('field_ui')) {
+      return;
+    }
+
+    $has_layout_builder = $this->moduleHandler->moduleExists('layout_builder');
```

Comment:
For now, and the foreseeable future, Layout Builder requires Field UI. So this is safe.